numba/_dispatcher.cpp

```diff
@@ -97,11 +98,12 @@ else                                                            \
     }                                                           \
 }
 
+typedef std::vector<Type> TypeTable;
+typedef std::vector<PyObject*> Functions;
 
-typedef struct DispatcherObject{
+class Dispatcher {
```

Comment:
Let's take the opportunity to document this code? Doc for class and methods?

Reply:
Documentation added.

Reply:
Thanks for doing that, looks good.